Validation/HGCalValidation/plugins/HGCalWaferValidation.cc

```diff
@@ -351,101 +422,64 @@ void HGCalWaferValidation::analyze(const edm::Event& iEvent, const edm::EventSet
-      else if (waferShapeStr == "bm")
-        waferShapeCode = HGCalTypes::WaferPartialType::WaferFive2;
-      else if (waferShapeStr == "c")
-        waferShapeCode = HGCalTypes::WaferPartialType::WaferThree;
-      else if (waferShapeStr == "d")
-        waferShapeCode = HGCalTypes::WaferPartialType::WaferSemi;
-      else if (waferShapeStr == "dm")
-        waferShapeCode = HGCalTypes::WaferPartialType::WaferSemi2;
-      else if (waferShapeStr == "g")
-        waferShapeCode = HGCalTypes::WaferPartialType::WaferChopTwo;
-      else if (waferShapeStr == "gm")
-        waferShapeCode = HGCalTypes::WaferPartialType::WaferChopTwoM;
-    }
+    const int waferLayer(std::stoi(tokens[0]));
+    const std::string waferShapeStr(tokens[1]);
+    const std::string waferDensityStr(isNewFile ? tokens[2].substr(0, 1) : "");
+    const int waferThickness(isNewFile ? std::stoi(tokens[2].substr(1)) : std::stoi(tokens[2]));
+    const double waferX(std::stod(tokens[3]));
+    const double waferY(std::stod(tokens[4]));
+    const int waferRotCode(std::stoi(tokens[5]));
+    const int waferU(std::stoi(tokens[6]));
+    const int waferV(std::stoi(tokens[7]));
+    const int waferShapeCode(
+        isNewFile ? (
+                        // if using new format flat file
+                        waferDensityStr == "l"   ? (waferShapeStr == "0"   ? HGCalTypes::WaferPartialType::WaferFull
+                                                    : waferShapeStr == "1" ? HGCalTypes::WaferPartialType::WaferHalf
+                                                    : waferShapeStr == "2" ? HGCalTypes::WaferPartialType::WaferHalf
+                                                    : waferShapeStr == "3" ? HGCalTypes::WaferPartialType::WaferSemi
+                                                    : waferShapeStr == "4" ? HGCalTypes::WaferPartialType::WaferSemi
+                                                    : waferShapeStr == "5" ? HGCalTypes::WaferPartialType::WaferFive
+                                                    : waferShapeStr == "6" ? HGCalTypes::WaferPartialType::WaferThree
+                                                                           : HGCalTypes::WaferPartialType::WaferOut)
+                        : waferDensityStr == "h" ? (waferShapeStr == "0"   ? HGCalTypes::WaferPartialType::WaferFull
+                                                    : waferShapeStr == "1" ? HGCalTypes::WaferPartialType::WaferHalf2
+                                                    : waferShapeStr == "2" ? HGCalTypes::WaferPartialType::WaferChopTwoM
+                                                    : waferShapeStr == "3" ? HGCalTypes::WaferPartialType::WaferSemi2
+                                                    : waferShapeStr == "4" ? HGCalTypes::WaferPartialType::WaferSemi2
+                                                    : waferShapeStr == "5" ? HGCalTypes::WaferPartialType::WaferFive2
```

Comment:
I am not sure if this kind of structures are oreferred over else ifs, I let experts to comment
@perrotta @qliphy

Reply:
Same comment applies here, both for the `if` and for the `map`.
Also, labelling evolution as `new` and `old` will tell us nothing the moment we have more than 2 versions. It would be better to either express is clearly with version numbers or use other labels.